consensus/src/liveness/leader_reputation.rs

```diff
@@ -217,6 +218,38 @@ impl NewBlockEventAggregation {
             .filter(move |&meta| meta.epoch() == epoch)
     }
 
+    pub fn get_aggregated_metrics(
+        &self,
+        epoch: u64,
+        candidates: &[Author],
+        history: &[NewBlockEvent],
+    ) -> (
+        HashMap<Author, u32>,
+        HashMap<Author, u32>,
+        HashMap<Author, u32>,
+    ) {
+        let votes = self.count_votes(epoch, candidates, history);
+        let proposals = self.count_proposals(epoch, history);
+        let failed_proposals = self.count_failed_proposals(epoch, candidates, history);
+
+        for candidate in candidates {
+            COMMITTED_PROPOSALS_IN_WINDOW
+                .with_label_values(&[candidate.short_str().as_str()])
+                .set(*proposals.get(candidate).unwrap_or(&0) as i64);
+            FAILED_PROPOSALS_IN_WINDOW
+                .with_label_values(&[candidate.short_str().as_str()])
+                .set(*failed_proposals.get(candidate).unwrap_or(&0) as i64);
+            COMMITTED_VOTES_IN_WINDOW
+                .with_label_values(&[candidate.short_str().as_str()])
+                .set(*votes.get(candidate).unwrap_or(&0) as i64);
+        }
+
+        LEADER_REPUTATION_HISTORY_SIZE.set(
+            proposals.values().sum::<u32>() as i64 + failed_proposals.values().sum::<u32>() as i64,
```

Comment:
@ikabiljo - IIUC there was a bug in this metrics calculation previously. Can you verify if this looks correct?

Reply:
@ikabiljo - The confusing part about this is when we only proposal failures for the epoch, the window size in previous implementation will be 0. I will rename this to  `LEADER_REPUTATION_HISTORY_SIZE_IN_ROUNDS` for clarity.